nodes/server.js

```diff
@@ -274,23 +279,34 @@ module.exports = function (RED) {
                     .then(circles => {
                         return circles;
                     })
+                    .catch(err => { node.handle_error(err) });
             });
         }
 
         getPeople(circleId, callback) {
             var node = this;
             
-            return life360.members(session, circleId).then(members => {
-                callback(members);
-            });
+            return life360.members(session, circleId)
+                .then(members => {
+                    callback(members);
+                })
+                .catch(err => { node.handle_error(err) });
         }
 
         getPlaces(circleId, callback) {
             var node = this;
 
-            return life360.places(session, circleId).then(places => {
-                callback(places);
-            });
+            return life360.places(session, circleId)
+                .then(places => {
+                    callback(places);
+                })
+                .catch(err => { node.handle_error(err) });
+        }
+
+        handle_error(err) {
+            let node = this;
+            node.error(err);
+            // node.status({fill:"red",shape:"ring",text:error});
```

Comment:
Generally, I'd say updating the status is a good thing, although since you pointed out that the server frequently has issues, we could keep an error count that gets zeroed out on every successful operation and then set a threshold after which to change the status to an error status... Though I'd be fine with a todo comment above this explaining why it's commented and referring to a feature issue on the GH repo.

Reply:
In addition, I wasn't sure if this was correct or not. I suspect this will update the server node status, and I am not sure if that is visible.

If you think you can get this working though, please do. Otherwise I can add a comment also. Or maybe just delete the line entirely.

Reply:
You can emit an error event from a server node and handle in in all location nodes and change their statuses. See [here](https://github.com/MaksimShakavin/node-red-contrib-life360/commit/69295af4a5418f565a07c73fbca3afcae8f01857)

Reply:
OK, so if I am reading this correctly, adding the following line will fix this?

```
node.emit('error', err.message);
```

Reply:
this will send an event. Then in location node you need to [subscribe](https://github.com/MaksimShakavin/node-red-contrib-life360/commit/69295af4a5418f565a07c73fbca3afcae8f01857#diff-83d32b2dfb71dc9ce73effd4164adad449738397b29e6fe22462783f40beb561R20) to this event, and change location node status. Then all location nodes will turn red, when server node gets an error 😉

Reply:
Is this particular issue resolved? If so, I will merge.

Reply:
I believe so, yes. Improvements are possible, but I think it is worth applying this patch as is for now.

Reply:
I've already made this improvements with status in my pr https://github.com/hepcat72/node-red-contrib-life360/pull/9
And I haven't merge this fixes into it, since I anyway rewrote most of the promises logic. So maybe it make sense to merge mine

Reply:
Assume you mean #10? If so, agreed, yours looks better. Removing deprecated libraries always good :-)